go-controller/pkg/ovn/policy.go

```diff
@@ -853,6 +853,14 @@ func (oc *Controller) handlePeerNamespaceAndPodSelector(
 				np.podHandlerList = append(np.podHandlerList, podHandler)
 			},
 			DeleteFunc: func(obj interface{}) {
+				// when the namespace labels no longer apply
+				// remove the namespaces pods from the address_set
+				namespace := obj.(*kapi.Namespace)
+				pods, _ := oc.kube.GetPods(namespace.Name, *podSelector)
```

Comment:
@JacobTanenbaum we should use the informer cache here rather than the kubeclient right?

Reply:
done